File: CodeTester-IntelliJ-IDEA/src/main/java/de/fxnm/action/BaseToggleAction.java
package de.fxnm.action;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.ToggleAction;
import com.intellij.openapi.project.Project;

import org.jetbrains.annotations.NotNull;

import java.util.Optional;

import static java.util.Optional.ofNullable;

public abstract class BaseToggleAction extends ToggleAction {

    protected Optional<Project> project(@NotNull final AnActionEvent event) {
        return ofNullable(getEventProject(event));
    }
}

File: CodeTester-IntelliJ-IDEA/src/main/kotlin/de/fxnm/action/BaseToggleAction.kt
package de.fxnm.action

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.ToggleAction
import com.intellij.openapi.project.Project
import java.util.Optional

abstract class BaseToggleAction : ToggleAction() {

    protected fun project(event: AnActionEvent): Optional<Project> {
        return Optional.ofNullable(getEventProject(event))
    }
}
